app/models/directory_item.rb

```diff
@@ -123,14 +130,13 @@ def self.refresh_period!(period_type, force: false)
         di.topic_count <> x.topic_count OR
         di.post_count <> x.post_count )
 
-      ",
-                  period_type: period_types[period_type],
-                  since: since,
-                  like_type: UserAction::LIKE,
-                  was_liked_type: UserAction::WAS_LIKED,
-                  new_topic_type: UserAction::NEW_TOPIC,
-                  reply_type: UserAction::REPLY,
-                  regular_post_type: Post.types[:regular]
+              ",
+              query_args
+             )
+
+      plugin_queries.each do |plugin_query|
```

Comment:
The best thing to do here would be to modify the query above to let plugins extend it. That gets _mega_ complicated really quickly. The extra queries shouldn't be too much of a strain on the server if they're written well, and this is already running as a background job.